### PR TITLE
Signal migration docs

### PR DESCRIPTION
**Pull request by agronholm**
## Changes

Adds a new section about migrating the signals and events code to Asphalt 5.

## Checklist

If this is a user-facing code change, like a bugfix or a new feature, please ensure that
you've fulfilled the following conditions (where applicable):

- [ ] You've added tests (in `tests/`) added which would fail without your patch
- [X] You've updated the documentation (in `docs/`, in case of behavior changes or new
features)
- [ ] You've added a new changelog entry (in `docs/versionhistory.rst`).

If this is a trivial change, like a typo fix or a code reformatting, then you can ignore
these instructions.

### Updating the changelog

If there are no entries after the last release, use `**UNRELEASED**` as the version.
If, say, your patch fixes issue <span>#</span>123, the entry should look like this:

```
- Fix big bad boo-boo in task groups
  (`#123 <https://github.com/asphalt-framework/asphalt/issues/123>`_; PR by @yourgithubaccount)
```

If there's no issue linked, just link to your pull request instead by updating the
changelog after you've created the PR.

### CODE REVIEW

File: docs/userguide/migration.rst [hunk 1]
```diff
@@ -287,6 +287,117 @@ After::
...
+++++++++++++++++++++++++++++++
+
+As the :class:`Event` class no longer has an initializer, you need to remove the
+``source`` and ``topic`` initializer parameters from your own subclasses, and drop the
+``super().__init__(source, topic)`` call. You may also want to take this opportunity to
+refactor them into data classes.
+
+Before::
+
+    from asphalt.core import Signal, Event
```
Comment (davidbrochart) [inline docs/userguide/migration.rst:307]:
```suggestion
    from asphalt.core import Event
```

File: docs/userguide/migration.rst [hunk 2]
```diff
@@ -287,6 +287,117 @@ After::
...
+    class MyEvent(Event):
+        def __init__(self, source, topic, an_attribute: str):
+            super().__init__(source, topic)
+            self.an_attribute = an_attribute
+
+After::
+
+    from dataclasses import dataclass
+
+    from asphalt.core import Signal, Event
```
Comment (davidbrochart) [inline docs/userguide/migration.rst:318]:
```suggestion
    from asphalt.core import Event
```

File: docs/userguide/migration.rst [hunk 3]
```diff
@@ -287,6 +287,117 @@ After::
...
+
+    @dataclass
+    class MyEvent(Event):
+        an_attribute: str
+
+Iterating over events
++++++++++++++++++++++
+
+As the ``connect()`` and ``disconnect()`` signal methods have been eliminated, you need
+to use the :meth:`Signal.stream_events` method or
```
Comment (davidbrochart) [inline docs/userguide/migration.rst:328]:
```suggestion
to use the :meth:`Signal.stream_events` method.
```

Comment (agronholm) [reply docs/userguide/migration.rst:328]: That unfinished sentence was actually meant to also mention the free `stream_events()` function

Comment (agronholm) [reply docs/userguide/migration.rst]: Updated.

**Review by davidbrochart — APPROVED** [review]

**Pull request merged**